Remove unused code in API::SerializedScriptValue after migration to JavaScriptEvaluationResult

achristensen07 opened the pull request:
#### 08912167893d71ada161221c29c441281976624a
<pre>
Remove unused code in API::SerializedScriptValue after migration to JavaScriptEvaluationResult
<a href="https://bugs.webkit.org/show_bug.cgi?id=289465">https://bugs.webkit.org/show_bug.cgi?id=289465</a>
<a href="https://rdar.apple.com/146664284">rdar://146664284</a>

Reviewed by Sihui Liu.

* Source/WebKit/Shared/API/APIObject.h:
* Source/WebKit/Shared/API/APISerializedScriptValue.cpp:
* Source/WebKit/Shared/API/APISerializedScriptValue.h:
(API::SerializedScriptValue::createFromWireBytes):
* Source/WebKit/UIProcess/API/Cocoa/APISerializedScriptValueCocoa.mm:
(API::SharedJSContext::singleton):
(API::SerializedScriptValue::deserialize): Deleted.
(API::validateObject): Deleted.
(API::coreValueFromNSObject): Deleted.
(API::SerializedScriptValue::createFromNSObject): Deleted.

Canonical link: <a href="https://commits.webkit.org/291906@main">https://commits.webkit.org/291906@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f45ceb2b948acae27278800da11d96cc8722cac4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94344 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13931 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3705 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99356 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44872 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14232 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22361 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/72002 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29338 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97346 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/10588 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/85200 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52324 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10279 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44190 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80514 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/2981 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101401 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21396 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15621 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/81009 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21648 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81206 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80368 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24923 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2302 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/14629 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/15141 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21380 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/26566 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21067 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/24527 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22808 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (approved):
r=me if bots are happy

achristensen07 committed:
Remove unused code in API::SerializedScriptValue after migration to JavaScriptEvaluationResult
https://bugs.webkit.org/show_bug.cgi?id=289465
rdar://146664284

Reviewed by Sihui Liu.

* Source/WebKit/Shared/API/APIObject.h:
* Source/WebKit/Shared/API/APISerializedScriptValue.cpp:
* Source/WebKit/Shared/API/APISerializedScriptValue.h:
(API::SerializedScriptValue::createFromWireBytes):
* Source/WebKit/UIProcess/API/Cocoa/APISerializedScriptValueCocoa.mm:
(API::SharedJSContext::singleton):
(API::SerializedScriptValue::deserialize): Deleted.
(API::validateObject): Deleted.
(API::coreValueFromNSObject): Deleted.
(API::SerializedScriptValue::createFromNSObject): Deleted.

Canonical link: https://commits.webkit.org/291906@main
